src/charm.py

```diff
@@ -642,580 +92,247 @@ def k8s_auth(self) -> None:
-        ingress_name = _INGRESS_SUB_REGEX.sub("-", hostname)
-        return f"{ingress_name}-ingress"
-
-    def _create_k8s_ingress_obj(
-        self, svc_hostname: str, initial_ingress: Any, paths: Any, label: str
-    ) -> Any:
-        """Create a Kubernetes Ingress Resources with the given data.
-
-        Args:
-            svc_hostname: Ingress service hostname.
-            initial_ingress: Initial ingress previously built.
-            paths: Ingress paths.
-            label: Ingress label.
-
-        Returns:
-            An Ingress object.
-        """
-        # Create a Ingress Object with the new ingress rules and return it.
-        rule = kubernetes.client.V1IngressRule(
-            host=svc_hostname,
-            http=kubernetes.client.V1HTTPIngressRuleValue(paths=paths),
-        )
-        # We're going to use an ingress name based on the desired service hostname.
-        ingress_name = self._ingress_name(svc_hostname)
-        new_spec = kubernetes.client.V1IngressSpec(rules=[rule])
-
-        # The service hostname might be configured to use TLS.
-        if initial_ingress.spec.tls:
-            new_spec.tls = [
-                kubernetes.client.V1IngressTLS(
-                    hosts=[svc_hostname],
-                    secret_name=initial_ingress.spec.tls[0].secret_name,
-                )
-            ]
-        return kubernetes.client.V1Ingress(
-            api_version="networking.k8s.io/v1",
-            kind="Ingress",
-            metadata=kubernetes.client.V1ObjectMeta(
-                name=ingress_name,
-                annotations=initial_ingress.metadata.annotations,
-                labels={CREATED_BY_LABEL: label},
-            ),
-            spec=new_spec,
-        )
-
-    def _define_ingress(self, body: Any) -> None:
-        """Create or update an ingress in kubernetes.
-
-        Args:
-            body: Ingress resource body to configure.
-        """
-        api = self._networking_v1_api()
-        self._look_up_and_set_ingress_class(api, body)
-        ingress_name = body.metadata.name
-        ingresses = api.list_namespaced_ingress(namespace=self._namespace)
-        if ingress_name in [x.metadata.name for x in ingresses.items]:
-            api.replace_namespaced_ingress(
-                name=ingress_name,
-                namespace=self._namespace,
-                body=body,
-            )
-            LOGGER.info(
-                "Ingress updated in namespace %s with name %s",
-                self._namespace,
-                ingress_name,
-            )
-        else:
-            api.create_namespaced_ingress(
-                namespace=self._namespace,
-                body=body,
-            )
-            LOGGER.info(
-                "Ingress created in namespace %s with name %s",
-                self._namespace,
-                ingress_name,
+        nginx_route_relations = self.model.relations["nginx-route"]
+        ingress_relations = self.model.relations["ingress"]
```

Comment:
I think for relations that have `limit: 1`, it is better to use `self.model.get_relation(RELATION_NAME)`, since that targets a single relation.

Reply:
That's a good idea, updated, thanks!